src/control/lib/daos/pool.go

```diff
@@ -87,8 +96,128 @@ type (
 		Free      uint64           `json:"free"`
 		MediaType StorageMediaType `json:"media_type"`
 	}
+
+	// PoolQueryMask implements a bitmask for pool query options.
+	PoolQueryMask C.uint64_t
+)
+
+const (
+	// DefaultPoolQueryMask defines the default pool query mask.
+	DefaultPoolQueryMask = PoolQueryMask(^uint64(0) &^ (C.DPI_ENGINES_ENABLED | C.DPI_ENGINES_DISABLED))
+	// HealthOnlyPoolQueryMask defines the mask for health-only queries.
+	HealthOnlyPoolQueryMask = PoolQueryMask(^uint64(0) &^ (C.DPI_ENGINES_ENABLED | C.DPI_SPACE))
+
+	// PoolQueryOptionSpace retrieves storage space usage as part of the pool query.
+	PoolQueryOptionSpace = "space"
+	// PoolQueryOptionRebuild retrieves pool rebuild status as part of the pool query.
+	PoolQueryOptionRebuild = "rebuild"
+	// PoolQueryOptionEnabledEngines retrieves enabled engines as part of the pool query.
+	PoolQueryOptionEnabledEngines = "enabled_engines"
+	// PoolQueryOptionDisabledEngines retrieves disabled engines as part of the pool query.
+	PoolQueryOptionDisabledEngines = "disabled_engines"
 )
 
+var poolQueryOptMap = map[C.int]string{
```

Comment:
Just to note that a lot of this mask handling duplicates the implementation in https://github.com/daos-stack/daos/blob/master/src/control/server/storage/config.go#L838 of `OptionBits` (where you offered lots of guidance). It might be beneficial to consolidate at some point so we have a single way of handling masks.

Reply:
> It might be beneficial to consolidate at some point so we have a single way of handling masks.

Good idea. I'd forgotten about that work.